#161340176 Enable social login using facebook & google

wycliffkas opened the pull request:
**What does this PR do?**
Adds Social Login using google and facebook

**Description of Task to be completed?**
Have the following endpoints working
**POST**  /api/facebook_login/
**POST** /api/google_login/

**How should this be manually tested?**
-After cloning the repo, setting up the virtual environment and installing dependencies.
-Use postman to test the above endpoints by adding a provider and access token in the body.
-access tokens can be obtained from [google oauthplayground](https://developers.google.com/oauthplayground/) and [facebook graph explorer](https://developers.facebook.com/tools/explorer/) for Google and Facebook respectively.

**What are the relevant pivotal tracker stories?**
[#161340176](https://www.pivotaltracker.com/story/show/161340176l)

<img width="1236" alt="screen shot 2018-11-16 at 16 27 24" src="https://user-images.githubusercontent.com/42246312/48624097-9fdd3d00-e9bc-11e8-9255-3a3edf21e9f1.png">
<img width="1238" alt="screen shot 2018-11-16 at 16 27 35" src="https://user-images.githubusercontent.com/42246312/48624098-a075d380-e9bc-11e8-86a2-4e6230a2b5d1.png">

SoniaRMK commented:
Kindly look into why the builds are failing. In your PR description, `How should this be manually tested` is lacking several steps. What happens after I run the server? Am I then automatically logged in by Facebook or Gmail? Also, squash your commits to only 1

SoniaRMK commented:
Resolve the merge conflicts as well

wycliffkas committed:
feat(API)Enable Social login using Google & Facebook

-implement login using a google account
-implement login using a facebook account

[Finishes #161340176]
